src/Formats/SchemaInferenceUtils.cpp

```diff
@@ -942,36 +927,22 @@ namespace
             /// and then as float.
             PeekableReadBuffer peekable_buf(buf);
             PeekableReadBufferCheckpoint checkpoint(peekable_buf);
+
+            if (tryReadFloat<is_json>(tmp_float, peekable_buf, settings, has_fractional) && has_fractional)
```

Comment:
```suggestion
            if (has_fractional && tryReadFloat<is_json>(tmp_float, peekable_buf, settings, has_fractional))
```

as it can fail quickly if bool is false